### PR TITLE
[Feature] Support query table function directly

### PR DESCRIPTION
**Pull request by sduzh**
Now we can query table functions like normal tables:
```
> select * from TABLE(unnest(ARRAY<INT>[1,2])) t(x); // select
+------+
| x    |
+------+
|    1 |
|    2 |
+------+
2 rows in set (0.02 sec)

> select * from t0; // t0 is a normal table
+------+
| c0   |
+------+
|    1 |
+------+
1 row in set (0.10 sec)

> select * from t0, TABLE(unnest(ARRAY<INT>[1, 2])); // cross join
+------+--------+
| c0   | unnest |
+------+--------+
|    1 |      1 |
|    1 |      2 |
+------+--------+
2 rows in set (0.03 sec)

> select * from t0 left join TABLE(unnest(ARRAY<INT>[1, 2])) t1(x) on t0.c0=t1.x; // left join
+------+------+
| c0   | x    |
+------+------+
|    1 |    1 |
+------+------+
1 row in set (0.06 sec)

> select * from t0 right join TABLE(unnest(ARRAY<INT>[1, 2])) t1(x) on t0.c0=t1.x; // right join
+------+------+
| c0   | x    |
+------+------+
|    1 |    1 |
| NULL |    2 |
+------+------+
2 rows in set (0.04 sec)
```
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java [hunk 1]
```diff
@@ -603,11 +603,19 @@ public LogicalPlan visitJoin(JoinRelation node, ExpressionMapping context) {
...
-                    node.getLeft().getRelationFields().joinWith(node.getRight().getRelationFields())),
-                    Streams.concat(leftPlan.getRootBuilder().getFieldMappings().stream(),
-                                    rightPlan.getRootBuilder().getFieldMappings().stream())
-                            .collect(Collectors.toList()));
+
+            List<ColumnRefOperator> leftFieldMappings = leftPlan.getRootBuilder().getFieldMappings();
+            List<ColumnRefOperator> rightFieldMappings = rightPlan.getRootBuilder().getFieldMappings();
+
+            ExpressionMapping expressionMapping;
+            if (node.getLeft().isDualRelation()) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java:611]: check right？

**Comment by Seaven** [comment]
add test case

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java]: Updated, PTAL

**Comment by sduzh** [comment]
I'll add some daily regression tests after this PR merged.

**Comment by imay** [comment]
> I'll add some daily regression tests after this PR merged.

why not add tests in our new sql-test framework?

**Comment by sduzh** [comment]
> > I'll add some daily regression tests after this PR merged.
> 
> why not add tests in our new sql-test framework?

I have considered it, but based on my communication with @wanpengfei-git , the sql-test framework has not been added to the CI for the time being.

**Comment by sduzh** [comment]
> > > I'll add some daily regression tests after this PR merged.
> > 
> > 
> > why not add tests in our new sql-test framework?
> 
> I have considered it, but based on my communication with @wanpengfei-git , the sql-test framework has not been added to the CI for the time being.

But anyway, I can add tests first.

**Comment by sduzh** [comment]
> > I'll add some daily regression tests after this PR merged.
> 
> why not add tests in our new sql-test framework?

Updated, all the test cases have passed on my local machine:
```
> python3 run.py --dir=sql/test_table_function  2>&1 | tee 1.log
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
Version: 2023-03-24 14:08:18.134582
Test cmd: nosetests test_sql_cases.py -vv -s --nologcapture --processes=8 --process-timeout=600

[DIR]: sql/test_table_function
[Mode]: VALIDATE
[file regex]: .*
[case regex]: .*
[attr]:

case num: 12
DB: ['test_db_bf02bc7e91694efca871f4b99a81e1c2']
[SQL]: select * from TABLE(unnest(ARRAY<INT>[1, NULL, 10]));
DB: ['test_db_e5182e2605db4de88c455704e79c48bc']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
DB: ['test_db_573cf2b2a4e34f71bf4bce24992e5a9f']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
DB: ['test_db_aba59d95a39e44f48e288ad7e3c36856']
[SQL]: select x from TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t(x);
DB: ['test_db_bcaeed0152054c998a7df0ac2ca84ae0']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[10, NULL, 100])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
DB: ['test_db_3a0f514db5f24e5b99c6f3a9313a39d3']
[SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) RIGHT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
DB: ['test_db_84ab6c749bc04a27abb0f5779a0db481']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[NULL, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
DB: ['test_db_dd4820a6a4a342639de0624a30e9ee01']
[SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) LEFT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
DB: ['test_db_a096ffb8f4924822b4be034cbbbbe991']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
DB: ['test_db_14001376514f42939f5330f0469f2d43']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1, 2, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
DB: ['test_db_69c7ec240f9742b4aa05839c33340f7d']
[SQL]: select x + 1 from (select * from TABLE(unnest(ARRAY<INT>[1], ARRAY<INT>[2, 3])) t(x, y)) t0;
DB: ['test_db_2252b825f4c94fa2873fe527a017f097']
[SQL]: create view v_unnest_subquery as select x + y from (select * from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[2, 3])) t1(x)) t2(x, y);
[SQL]: select * from v_unnest_subquery;
0    : sql/test_table_function/R/test_unnest:test_select ... ok
1    : sql/test_table_function/R/test_unnest:test_alias ... ok
3    : sql/test_table_function/R/test_unnest:test_cross_join3 ... ok
2    : sql/test_table_function/R/test_unnest:test_cross_join4 ... ok
5    : sql/test_table_function/R/test_unnest:test_cross_join2 ... ok
4    : sql/test_table_function/R/test_unnest:test_cross_join ... ok
9    : sql/test_table_function/R/test_unnest:test_subquery ... ok
7    : sql/test_table_function/R/test_unnest:test_cross_join_conditional ... ok
6    : sql/test_table_function/R/test_unnest:test_cross_join_conditional2 ... ok
11   : sql/test_table_function/R/test_unnest:test_view ... ok
8    : sql/test_table_function/R/test_unnest:test_left_join ... ok
10   : sql/test_table_function/R/test_unnest:test_right_join ... ok

----------------------------------------------------------------------
Ran 12 tests in 0.118s

OK
```

**Comment by sduzh** [comment]
> > > I'll add some daily regression tests after this PR merged.
> > 
> > 
> > why not add tests in our new sql-test framework?
> 
> Updated, all the test cases have passed on my local machine:
> 
> ```
> > python3 run.py --dir=sql/test_table_function  2>&1 | tee 1.log
> nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
> Version: 2023-03-24 14:08:18.134582
> Test cmd: nosetests test_sql_cases.py -vv -s --nologcapture --processes=8 --process-timeout=600
> 
> [DIR]: sql/test_table_function
> [Mode]: VALIDATE
> [file regex]: .*
> [case regex]: .*
> [attr]:
> 
> case num: 12
> DB: ['test_db_bf02bc7e91694efca871f4b99a81e1c2']
> [SQL]: select * from TABLE(unnest(ARRAY<INT>[1, NULL, 10]));
> DB: ['test_db_e5182e2605db4de88c455704e79c48bc']
> [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> DB: ['test_db_573cf2b2a4e34f71bf4bce24992e5a9f']
> [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
> DB: ['test_db_aba59d95a39e44f48e288ad7e3c36856']
> [SQL]: select x from TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t(x);
> DB: ['test_db_bcaeed0152054c998a7df0ac2ca84ae0']
> [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[10, NULL, 100])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> DB: ['test_db_3a0f514db5f24e5b99c6f3a9313a39d3']
> [SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) RIGHT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
> DB: ['test_db_84ab6c749bc04a27abb0f5779a0db481']
> [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[NULL, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> DB: ['test_db_dd4820a6a4a342639de0624a30e9ee01']
> [SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) LEFT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
> DB: ['test_db_a096ffb8f4924822b4be034cbbbbe991']
> [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> DB: ['test_db_14001376514f42939f5330f0469f2d43']
> [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1, 2, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
> DB: ['test_db_69c7ec240f9742b4aa05839c33340f7d']
> [SQL]: select x + 1 from (select * from TABLE(unnest(ARRAY<INT>[1], ARRAY<INT>[2, 3])) t(x, y)) t0;
> DB: ['test_db_2252b825f4c94fa2873fe527a017f097']
> [SQL]: create view v_unnest_subquery as select x + y from (select * from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[2, 3])) t1(x)) t2(x, y);
> [SQL]: select * from v_unnest_subquery;
> 0    : sql/test_table_function/R/test_unnest:test_select ... ok
> 1    : sql/test_table_function/R/test_unnest:test_alias ... ok
> 3    : sql/test_table_function/R/test_unnest:test_cross_join3 ... ok
> 2    : sql/test_table_function/R/test_unnest:test_cross_join4 ... ok
> 5    : sql/test_table_function/R/test_unnest:test_cross_join2 ... ok
> 4    : sql/test_table_function/R/test_unnest:test_cross_join ... ok
> 9    : sql/test_table_function/R/test_unnest:test_subquery ... ok
> 7    : sql/test_table_function/R/test_unnest:test_cross_join_conditional ... ok
> 6    : sql/test_table_function/R/test_unnest:test_cross_join_conditional2 ... ok
> 11   : sql/test_table_function/R/test_unnest:test_view ... ok
> 8    : sql/test_table_function/R/test_unnest:test_left_join ... ok
> 10   : sql/test_table_function/R/test_unnest:test_right_join ... ok
> 
> ----------------------------------------------------------------------
> Ran 12 tests in 0.118s
> 
> OK
> ```

Sorry, there is an unexpected result, I'm working on it.

**Comment by sduzh** [comment]
> > > > I'll add some daily regression tests after this PR merged.
> > > 
> > > 
> > > why not add tests in our new sql-test framework?
> > 
> > 
> > Updated, all the test cases have passed on my local machine:
> > ```
> > > python3 run.py --dir=sql/test_table_function  2>&1 | tee 1.log
> > nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
> > Version: 2023-03-24 14:08:18.134582
> > Test cmd: nosetests test_sql_cases.py -vv -s --nologcapture --processes=8 --process-timeout=600
> > 
> > [DIR]: sql/test_table_function
> > [Mode]: VALIDATE
> > [file regex]: .*
> > [case regex]: .*
> > [attr]:
> > 
> > case num: 12
> > DB: ['test_db_bf02bc7e91694efca871f4b99a81e1c2']
> > [SQL]: select * from TABLE(unnest(ARRAY<INT>[1, NULL, 10]));
> > DB: ['test_db_e5182e2605db4de88c455704e79c48bc']
> > [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> > DB: ['test_db_573cf2b2a4e34f71bf4bce24992e5a9f']
> > [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
> > DB: ['test_db_aba59d95a39e44f48e288ad7e3c36856']
> > [SQL]: select x from TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t(x);
> > DB: ['test_db_bcaeed0152054c998a7df0ac2ca84ae0']
> > [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[10, NULL, 100])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> > DB: ['test_db_3a0f514db5f24e5b99c6f3a9313a39d3']
> > [SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) RIGHT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
> > DB: ['test_db_84ab6c749bc04a27abb0f5779a0db481']
> > [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[NULL, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> > DB: ['test_db_dd4820a6a4a342639de0624a30e9ee01']
> > [SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) LEFT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
> > DB: ['test_db_a096ffb8f4924822b4be034cbbbbe991']
> > [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> > DB: ['test_db_14001376514f42939f5330f0469f2d43']
> > [SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1, 2, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
> > DB: ['test_db_69c7ec240f9742b4aa05839c33340f7d']
> > [SQL]: select x + 1 from (select * from TABLE(unnest(ARRAY<INT>[1], ARRAY<INT>[2, 3])) t(x, y)) t0;
> > DB: ['test_db_2252b825f4c94fa2873fe527a017f097']
> > [SQL]: create view v_unnest_subquery as select x + y from (select * from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[2, 3])) t1(x)) t2(x, y);
> > [SQL]: select * from v_unnest_subquery;
> > 0    : sql/test_table_function/R/test_unnest:test_select ... ok
> > 1    : sql/test_table_function/R/test_unnest:test_alias ... ok
> > 3    : sql/test_table_function/R/test_unnest:test_cross_join3 ... ok
> > 2    : sql/test_table_function/R/test_unnest:test_cross_join4 ... ok
> > 5    : sql/test_table_function/R/test_unnest:test_cross_join2 ... ok
> > 4    : sql/test_table_function/R/test_unnest:test_cross_join ... ok
> > 9    : sql/test_table_function/R/test_unnest:test_subquery ... ok
> > 7    : sql/test_table_function/R/test_unnest:test_cross_join_conditional ... ok
> > 6    : sql/test_table_function/R/test_unnest:test_cross_join_conditional2 ... ok
> > 11   : sql/test_table_function/R/test_unnest:test_view ... ok
> > 8    : sql/test_table_function/R/test_unnest:test_left_join ... ok
> > 10   : sql/test_table_function/R/test_unnest:test_right_join ... ok
> > 
> > ----------------------------------------------------------------------
> > Ran 12 tests in 0.118s
> > 
> > OK
> > ```
> 
> Sorry, there is an unexpected result in the file `test/sql/test_table_function/R/test_unnest`, I'm working on it:
> 
> ```
> -- name: test_cross_join3
> select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
> -- result:
> E: (1064, 'Unexpected exception: null')
> -- !result
> ```

This was caused by an unrelated issue https://github.com/StarRocks/starrocks/issues/20215, I've remove the
this test case.

**Comment by sduzh** [comment]
All tests passed on my local machine:
```
➜  test git:(feature/select-table-function) ✗ python3 run.py --dir=sql/test_table_function  2>&1 | tee 1.log
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
Version: 2023-03-24 14:28:55.815929
Test cmd: nosetests test_sql_cases.py -vv -s --nologcapture --processes=8 --process-timeout=600

[DIR]: sql/test_table_function
[Mode]: VALIDATE
[file regex]: .*
[case regex]: .*
[attr]:

case num: 11
DB: ['test_db_fd36fd2f656545d688175284e190da12']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1, 2, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
DB: ['test_db_f269414e099c4e36906dbd1fd49e08ed']
[SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) RIGHT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
DB: ['test_db_d5d3df3c24f2405fa8f92198e1de18be']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[NULL, NULL])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
DB: ['test_db_459f36f1238a4473991af4dbcb5147c0']
[SQL]: select * from TABLE(unnest(ARRAY<INT>[1, NULL, 10]));
DB: ['test_db_e12f49d157fe4744b4aef79cdd3f3a46']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[10, NULL, 100])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
DB: ['test_db_e84630b7d18d4a2883e4bd0ad2486d24']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x);
DB: ['test_db_f2df6179710a400faa1c2c82373a25e8']
[SQL]: select x from TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t(x);
DB: ['test_db_fcbe2755440c4dcca401845f65061482']
[SQL]: select x + 1 from (select * from TABLE(unnest(ARRAY<INT>[1], ARRAY<INT>[2, 3])) t(x, y)) t0;
DB: ['test_db_218a76a2de3d4175b599928ab6e09207']
[SQL]: select * from TABLE(unnest(ARRAY<INT>[1])) t0(x) LEFT JOIN TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) on t0.x=t1.x;
DB: ['test_db_f177d8408e5942f6b22375dd8a3955e1']
[SQL]: select t0.x, t1.x from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[1, NULL, 10])) t1(x) where t0.x=t1.x;
DB: ['test_db_2bd0121fd1c04764b2e2acf9cc901071']
[SQL]: create view v_unnest_subquery as select x + y from (select * from TABLE(unnest(ARRAY<INT>[1])) t0(x), TABLE(unnest(ARRAY<INT>[2, 3])) t1(x)) t2(x, y);
[SQL]: select * from v_unnest_subquery;
1    : sql/test_table_function/R/test_unnest:test_cross_join4 ... ok
2    : sql/test_table_function/R/test_unnest:test_cross_join2 ... ok
3    : sql/test_table_function/R/test_unnest:test_alias ... ok
0    : sql/test_table_function/R/test_unnest:test_cross_join_conditional2 ... ok
5    : sql/test_table_function/R/test_unnest:test_select ... ok
6    : sql/test_table_function/R/test_unnest:test_cross_join ... ok
8    : sql/test_table_function/R/test_unnest:test_subquery ... ok
4    : sql/test_table_function/R/test_unnest:test_left_join ... ok
10   : sql/test_table_function/R/test_unnest:test_view ... ok
9    : sql/test_table_function/R/test_unnest:test_right_join ... ok
7    : sql/test_table_function/R/test_unnest:test_cross_join_conditional ... ok

----------------------------------------------------------------------
Ran 11 tests in 0.127s

OK
```

**Review by Seaven — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -1629,11 +1629,15 @@ relationPrimary
     | '(' VALUES rowConstructor (',' rowConstructor)* ')'
         (AS? alias=identifier columnAliases?)?                                          #inlineTable
     | subquery (AS? alias=identifier columnAliases?)?                                   #subqueryWithAlias
-    | qualifiedName '(' expressionList ')'
-        (AS? alias=identifier columnAliases?)?                                          #tableFunction
+    | tableFunctionCall (AS? alias=identifier columnAliases?)?                          #tableFunction
+    | TABLE '(' tableFunctionCall ')' (AS? alias=identifier columnAliases?)?            #tableFunctionTable
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1633]: Can you unify tableFunctionTable and NormalizedTableFunctionRelation into one name?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 3]
```diff
@@ -1629,11 +1629,15 @@ relationPrimary
...
     | '(' relations ')'                                                                 #parenthesizedRelation
     ;
 
+tableFunctionCall
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1637]: I do not recommend adding a tableFunctionCall. Although this logic is not wrong. However, tableFunctionCall and FunctionCallExpr are not in one-to-one correspondence, which can easily cause trouble for other developers. I think it is enough to write directly, TABLE '(' qualifiedName '(' expressionList ')' ')' (AS? alias=identifier columnAliases?)?

**Review by HangyuanLiu — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 3 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1637]: ditto

**Pull request merged**